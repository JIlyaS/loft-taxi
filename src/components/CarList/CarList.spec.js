import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import {CarList} from './';

const carList = [
  {
    name: 'Стандарт',
    cost: 150,
    type: 'standart',
    checked: true,
  },
  {
    name: 'Премиум',
    cost: 250,
    type: 'premium',
    checked: false,
  },
  {
    name: 'Бизнес',
    cost: 300,
    type: 'business',
    checked: false,
  },
];

describe('CarList', () => {

  it('renders correctly snapshot', () => {
    const tree = renderer.create(<CarList carList={carList} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render correct', () => {
    const { queryByTestId } = render(<CarList carList={carList} />);
    expect(queryByTestId('car-list')).toBeInTheDocument();
    expect(queryByTestId('car-list-standart')).toBeInTheDocument();
    expect(queryByTestId('car-list-premium')).toBeInTheDocument();
    expect(queryByTestId('car-list-business')).toBeInTheDocument();
  });
});
