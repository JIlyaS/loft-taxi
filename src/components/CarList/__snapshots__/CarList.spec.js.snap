// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarList renders correctly snapshot 1`] = `
<div
  className="car-list"
  data-testid="car-list"
>
  <input
    checked={true}
    className="car-list__radio"
    id="car-list-standart"
    name="standart"
    type="radio"
    value="standart"
  />
  <label
    className="car-list__label"
    data-testid="car-list-standart"
    htmlFor="car-list-standart"
  >
    <div
      className="MuiPaper-root MuiCard-root car-list__item MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root car-list__content"
      >
        <h2
          className="car-list__title"
        >
          Стандарт
        </h2>
        <span
          className="car-list__cost-name"
        >
          Стоимость
        </span>
        <span
          className="car-list__cost"
        >
          150
        </span>
        <img
          alt="Стандарт такси"
          className="car-list__img car-list__img--standart"
          height="73px"
          src="car-standart.svg"
          width="95px"
        />
      </div>
    </div>
  </label>
  <input
    checked={false}
    className="car-list__radio"
    id="car-list-premium"
    name="premium"
    type="radio"
    value="premium"
  />
  <label
    className="car-list__label"
    data-testid="car-list-premium"
    htmlFor="car-list-premium"
  >
    <div
      className="MuiPaper-root MuiCard-root car-list__item MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root car-list__content"
      >
        <h2
          className="car-list__title"
        >
          Премиум
        </h2>
        <span
          className="car-list__cost-name"
        >
          Стоимость
        </span>
        <span
          className="car-list__cost"
        >
          250
        </span>
        <img
          alt="Премиум такси"
          className="car-list__img car-list__img--premium"
          height="81px"
          src="car-premium.svg"
          width="118px"
        />
      </div>
    </div>
  </label>
  <input
    checked={false}
    className="car-list__radio"
    id="car-list-business"
    name="business"
    type="radio"
    value="business"
  />
  <label
    className="car-list__label"
    data-testid="car-list-business"
    htmlFor="car-list-business"
  >
    <div
      className="MuiPaper-root MuiCard-root car-list__item MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root car-list__content"
      >
        <h2
          className="car-list__title"
        >
          Бизнес
        </h2>
        <span
          className="car-list__cost-name"
        >
          Стоимость
        </span>
        <span
          className="car-list__cost"
        >
          300
        </span>
      </div>
    </div>
  </label>
</div>
`;
